src/molecule/config.py

```diff
@@ -230,7 +230,7 @@ def _get_driver_name(self):
         driver_from_state_file = self.state.driver
         driver_from_cli = self.command_args.get("driver_name")
 
-        if driver_from_state_file:
+        if driver_from_state_file and driver_from_state_file in api.drivers():
```

Comment:
maybe I'm seeing problems where there are none because I've not understood what it's doing... I'm wondering if the check is enough. For instance, the scenario is configured to use driver "foo", the user installs driver "bar" and modify the configuration to use driver "bar" but doesn't uninstall the driver "foo", will the driver change get noticed ?

Reply:
I think those are two different problems. "We are missing a driver that initialized scenario is using" is one and the other is "Scenario and its initialized instance diverged."

Current PR is trying to solve the first issue (making sure we can still converge, etc., against an initialized scenario) and does not tackle the second issue.

I am not sure if it is even possible to solve the second problem without making some severe changes to Molecule's internals. Why? It seems to me that once we initialize the scenario, Molecule should ignore changes to the source configuration and always use configuration from the initialized instance. But currently, Molecule does not distinguish this and always uses source config as a fallback configuration.